zml/module.zig

```diff
@@ -1157,29 +763,38 @@ fn computeModuleHash(platform: Platform, module: mlir.Module) u64 {
     return hasher.final();
 }
 
-const max_pjrt_executable_size = 400 * 1024 * 1024;
+fn absoluteCacheFileZ(arena: std.mem.Allocator, cache_path: ?[]const u8, module_hash: u64) !?[:0]const u8 {
+    if (cache_path == null) return null;
+    const resolved_path = try std.fs.cwd().realpathAlloc(arena, cache_path.?);
+    std.fs.makeDirAbsolute(resolved_path) catch |err| switch (err) {
+        error.PathAlreadyExists => {},
+        else => return err,
+    };
 
-fn loadPjrtExecutable(arena: std.mem.Allocator, platform: Platform, module_hash: u64, compilation_cache_location: []const u8) !*pjrt.LoadedExecutable {
-    const resolved_path = try std.fs.cwd().realpathAlloc(arena, compilation_cache_location);
-    const compilation_cache_dir = try std.fs.openDirAbsolute(resolved_path, .{});
-    var buf: [16]u8 = undefined;
-    const filename = try std.fmt.bufPrint(&buf, "{x}", .{module_hash});
-    const loaded_executable_file = try compilation_cache_dir.openFile(filename, .{});
+    var buf: [24]u8 = undefined;
+    const module_name = std.fmt.bufPrint(&buf, "{x}.pjrt", .{module_hash}) catch unreachable;
+    return try std.fs.path.joinZ(arena, &.{ resolved_path, module_name });
+}
+
+fn loadPjrtExecutable(arena: std.mem.Allocator, platform: Platform, absolute_file: [:0]const u8) !*pjrt.LoadedExecutable {
+    const loaded_executable_file = try std.fs.openFileAbsoluteZ(absolute_file, .{});
     defer loaded_executable_file.close();
 
-    const bytes = try loaded_executable_file.readToEndAlloc(arena, max_pjrt_executable_size);
+    const exe_size = if (loaded_executable_file.stat()) |stat| stat.size else |_| 400 * 1024 * 1024;
```

Comment:
Can you restore the `max_pjrt_executable_size` constant ? So that we know where the magic number comes from ?

Reply:
ok